### PR TITLE
Rest client 2.0 updates

### LINKED ISSUE #4692: mp-rest-client ignores SSL configuration
https://github.com/eclipse-ee4j/jersey/commit/bc67ee3a47b2da87fe8ba6aab11bcd82d458b278#r46079691

When configuring MP REST Client:
```
book-service/mp-rest/url: https://127.0.0.1:8480
book-service/mp-rest/trustStore: classpath:keystore.p12
book-service/mp-rest/trustStoreType: PKCS12
book-service/mp-rest/trustStorePassword: password
book-service/mp-rest/hostnameVerifier: org.org.AcceptAnyHostname
book-service/mp-rest/providers:
   - org.glassfish.jersey.netty.connector.NettyConnectorProvider
```
The TLS configuration is totally ignored. This is caused by the routine that applies the configuration is in `RestClientBuilderImpl`.

This seems to be the correct way, and correct order:

```
diff --git a/ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientBuilderImpl.java b/ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jers
ey/microprofile/restclient/RestClientBuilderImpl.java
index 16abf1fae..c1e1e58d1 100644
--- a/ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientBuilderImpl.java
+++ b/ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientBuilderImpl.java
@@ -160,37 +160,52 @@ class RestClientBuilderImpl implements RestClientBuilder {
         //sort all AsyncInvocationInterceptorFactory by priority
         asyncInterceptorFactories.sort(Comparator.comparingInt(AsyncInvocationInterceptorFactoryPriorityWrapper::getPriority));
 
+        if (connector != null) {
+            ClientConfig config = new ClientConfig();
+            config.loadFrom(getConfiguration());
+            config.connectorProvider(connector);
+            clientBuilder = clientBuilder.withConfig(config); // apply config...
+        }
+       // override ClientConfig with values that have been set explicitly
         clientBuilder.executorService(new ExecutorServiceWrapper(executorService.get()));

         if (null != sslContext) {
             clientBuilder.sslContext(sslContext);
         }
 
         if (null != sslHostnameVerifier) {
             clientBuilder.hostnameVerifier(sslHostnameVerifier);
         }
 
         if (null != sslTrustStore) {
             clientBuilder.trustStore(sslTrustStore);
         }
 
         if (null != sslKeyStore) {
             clientBuilder.keyStore(sslKeyStore, sslKeyStorePassword);
         }
 
-        Client client;
-        if (connector == null) {
-            client = clientBuilder.build();
-        } else {
-            ClientConfig config = new ClientConfig();
-            config.loadFrom(getConfiguration());
-            config.connectorProvider(connector);
-            client = ClientBuilder.newClient(config);
-        }
+        Client client = clientBuilder.build();
 
         if (client instanceof Initializable) {
             ((Initializable) client).preInitialize();
         }
```

### LINKED ISSUE #4715: Rest client does not properly add headers from ClientHeadersFactory
Headers returned by the ClientHeadersFactory are not properly added to the request headers.

### LINKED ISSUE #4717: Add tests for newly updated Jersey classes by RestClient
- [ ] Add tests for newly updated Jersey classes by RestClient
- [ ] Fix some minor functional problems

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #4692
Fixes #4715
Fixes #4717

### CODE REVIEW

File: ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientModel.java [hunk 1]
```diff
@@ -46,28 +46,11 @@
     /**
      * Creates new instance of the {@link RestClientModel} base on interface class.
      *
-     * @param restClientClass           rest client interface
-     * @param responseExceptionMappers  registered exception mappers
-     * @param paramConverterProviders   registered param converters
-     * @param inboundHeadersProviders   registered inbound header providers
-     * @param asyncInterceptorFactories registered async interceptor factories
-     * @param injectionManager
+     * @param context RestClient data context
```
Comment (jansupol) [inline ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientModel.java:49]: Please update copyright year

File: core-client/src/main/java/org/glassfish/jersey/client/JerseyWebTarget.java [hunk 2]
```diff
@@ -149,10 +149,9 @@ public JerseyWebTarget matrixParam(String name, Object... values) throws NullPoi
     public JerseyWebTarget queryParam(String name, Object... values) throws NullPointerException {
         checkNotClosed();
         UriBuilder uriBuilder = getUriBuilder();
-        if (uriBuilder instanceof JerseyUriBuilder) {
-            ((JerseyUriBuilder) uriBuilder).setQueryParamStyle((JerseyQueryParamStyle) this.getConfiguration()
-                            .getProperty(ClientProperties.QUERY_PARAM_STYLE)
-            );
+        Object queryParamProperty = this.getConfiguration().getProperty(ClientProperties.QUERY_PARAM_STYLE);
+        if (queryParamProperty != null && uriBuilder instanceof JerseyUriBuilder) {
+            ((JerseyUriBuilder) uriBuilder).setQueryParamStyle((JerseyQueryParamStyle) queryParamProperty);
```
Comment (jansupol) [inline core-client/src/main/java/org/glassfish/jersey/client/JerseyWebTarget.java:154]: Would it be better if we had a check whether the queryParamProperty is JerseyQueryParamStyle?

Comment (Verdent) [reply core-client/src/main/java/org/glassfish/jersey/client/JerseyWebTarget.java:154]: changed

File: ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientModel.java (hunk 1 above)
Comment (Verdent) [reply ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/RestClientModel.java:49]: updated

**Review by jansupol — APPROVED** [review]

**Review by senivam — APPROVED** [review]

**Pull request merged**